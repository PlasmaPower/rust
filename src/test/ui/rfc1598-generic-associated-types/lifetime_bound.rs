// check-pass

// rust-lang/rust#62521: Do not ICE when a generic lifetime is used
// in the type's bound.

#![feature(generic_associated_types)]
//~^ WARNING the feature `generic_associated_types` is incomplete

trait Foo {
  type PublicKey<'a>: From<&'a [u8]>;
}

trait Bar {
    type Item<'a>: 'a;
}

fn main() {}
