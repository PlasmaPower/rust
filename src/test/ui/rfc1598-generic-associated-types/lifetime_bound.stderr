warning: the feature `generic_associated_types` is incomplete and may cause the compiler to crash
  --> $DIR/lifetime_bound.rs:6:12
   |
LL | #![feature(generic_associated_types)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default

